packages/dashboard-frontend/src/services/workspace-client/devworkspace/DevWorkspaceEditorProcessTheia.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2018-2021 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { InversifyBinding } from '@eclipse-che/che-theia-devworkspace-handler/lib/inversify/inversify-binding';
+import { CheTheiaPluginsDevfileResolver } from '@eclipse-che/che-theia-devworkspace-handler/lib/devfile/che-theia-plugins-devfile-resolver';
+import common from '@eclipse-che/common';
+import { injectable } from 'inversify';
+import { IDevWorkspaceEditorProcess, IDevWorkspaceEditorProcessingContext } from './devWorkspaceClient';
+
+/**
+ * This class manages what needs to be added on devfile/devworkspace template for Che-Theia
+ */
+@injectable()
+export class DevWorkspaceEditorProcessTheia implements IDevWorkspaceEditorProcess {
+
+    /**
+     * Match if the editor metadata name contains the name theia
+     */
+    match(context: IDevWorkspaceEditorProcessingContext): boolean {
+        return context.editorsDevfile.some(editor => editor.metadata.name.toLowerCase().includes('theia'));
```

Comment:
that again could leave on the editor lib side but here I have a weaker feeling that it should.